src/pages/index.js

```diff
@@ -17,51 +19,127 @@ import {
+    authorization: "7ed3bb2f-7b5a-422d-a884-e2fa31c7bd62",
+    "Content-Type": "application/json",
+  },
+});
 
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
 });
-const editProfilePopup = new PopupWithForm(
-  "#profile-edit-modal",
-  ({ name, description }) => handleEditProfileSubmit(name, description)
-);
 
-const addCardPopup = new PopupWithForm("#add-card-modal", ({ title, link }) => {
-  handleAddCardSubmit(title, link);
+const editAvatarPopup = new PopupWithForm("#modal-edit-avatar", (data) => {
+  return api
+    .updateAvatar(data)
+    .then((res) => {
+      userInfo.setUserInfo(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
+editAvatarPopup.setEventListeners();
+
+document
+  .querySelector("#profile-image-pencil")
+  .addEventListener("click", () => {
+    formValidators["edit-avatar-form"].resetValidation();
+    editAvatarPopup.open();
+  });
+
+const confirmAction = (card, cardId) => {
+  return api
+    .deleteCard(cardId)
+    .then(() => {
+      card.removeCard();
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+};
+
+const deleteCardPopup = new PopupWithConfirmation({
+  popupSelector: "#delete-card-modal",
+  handleConfirm: confirmAction,
 });
+
+deleteCardPopup.setEventListeners();
+
 const iamgePreviewPopup = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 
-/* ----------------------- */
-/*      Event Listner      */
-/* ----------------------- */
-// eneble event listeners in each form
-editProfilePopup.setEventListeners();
-addCardPopup.setEventListeners();
-iamgePreviewPopup.setEventListeners();
-
-const renderCard = (cardData) => {
-  const newCard = new Card(
-    cardData,
-    cardSelector,
-    iamgePreviewPopup,
-    (title, link) => iamgePreviewPopup.open(title, link)
+let section;
+
+const renderCard = (data) => {
+  const cardElement = new Card(
+    {
+      data: data,
+      handleImageClick: (imageData) => {
+        iamgePreviewPopup.open(imageData);
+      },
+      handleDeleteCard: (card, cardId) => {
+        deleteCardPopup.open(card, cardId);
+      },
+      confirmPopup: deleteCardPopup,
+      api: api,
+    },
+    "#card-template"
   );
-  section.addItem(newCard.getView());
-};
 
-const section = new Section(
+  section.addItem(cardElement.getView());
+};
+section = new Section(
   {
-    items: initialCards,
     renderer: renderCard,
   },
-
   cardsListSelector
 );
 
+const addCardPopup = new PopupWithForm("#add-card-modal", (data) => {
+  return api
+    .addNewCards(data)
+    .then((res) => {
+      renderCard(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
+
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, cards]) => {
+    userInfo.setUserInfo(userData);
+    section.renderItems(cards);
+  })
+  .catch((error) => {
+    console.error(error);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors